proxy/api/src/context.rs

```diff
@@ -212,39 +205,31 @@ impl Unsealed {
         let store = kv::Store::new(kv::Config::new(tmp_dir.path().join("store")))?;
 
         let key = link_crypto::SecretKey::new();
-        let signer = BoxedSigner::from(link_crypto::SomeSigner { signer: key });
         let paths = librad::paths::Paths::from_root(tmp_dir.path())?;
 
-        let (peer_control, peer, run_handle) = {
-            let config = radicle_daemon::config::default(signer, tmp_dir.path())?;
-            let disco = radicle_daemon::config::static_seed_discovery(&[]);
-            let coco_peer = radicle_daemon::Peer::new(
-                config,
-                disco,
-                store.clone(),
-                radicle_daemon::RunConfig::default(),
-            )?;
-            let peer = coco_peer.peer.clone();
-
-            let peer_control = coco_peer.control();
-            let run_handle = async move {
-                let (shutdown, run) = coco_peer.start();
-                // We spawn a task for `run` so that it is run to completion even if this future is
-                // dropped.
-                let run = tokio::task::spawn(run);
-                if let Err(err) = run.await {
-                    tracing::error!(?err, "peer run error");
-                }
-                drop(shutdown);
-            };
-            (peer_control, peer, run_handle)
+        let seeds_watch = tokio::sync::watch::channel(vec![]).1;
+
+        let (peer, peer_runner) = crate::peer::create(crate::peer::Config {
+            paths: paths.clone(),
+            key,
+            store: store.clone(),
+            discovery: radicle_daemon::config::StreamDiscovery::new(seeds_watch),
+            listen: "127.0.0.1:0".parse().expect("invalid IP address"),
```

Comment:
Why is this the loopback address?

Reply:
The test peer should not be exposed to the outside world. This is the same as with the test daemon configuration: https://github.com/radicle-dev/radicle-link/blob/258c893bf833b7ce37569c7fed4d3f6746ffe0a7/daemon/src/config.rs#L47

Reply:
Ah, missed the `#[cfg(test)]`.